src/application/components/Explorer/Explorer.tsx

```diff
@@ -43,20 +44,21 @@ function executeOperation({
 }: {
   operation: string;
   operationName?: string;
-  variables?: JSONValue;
+  variables?: JSONObject;
   fetchPolicy: FetchPolicy;
   isSubscription?: boolean;
 }) {
   return new Observable<QueryResult>((observer) => {
-    const payload = JSON.stringify({
-      operation,
-      operationName,
-      variables,
-      fetchPolicy,
+    panelWindow.send({
+      type: "explorerRequest",
+      payload: {
+        operation: gql(operation),
+        operationName,
+        variables,
+        fetchPolicy,
+      },
```

Comment:
Actually, no need to `stringify` this one.

Everything here already came in from a `postMessage`, so it's transportable. The only possible exception would be the `DocumentNode` itself, but that seems to work nicely with `structuredClone`.